### DISCUSSION TITLE
Retrain 2D U-Net: sparse annotations

### DISCUSSION THREAD

**Discussion by MirenLurBarquin** [ivadomed/ivadomed#1324] [Q&A]
Dear all, 

I am currently using [AxonDeepSeg](https://axondeepseg.readthedocs.io/en/latest/#) and [Ivadomed](https://ivadomed.org/tutorials/two_class_microscopy_seg_2d_unet.html) for my project; however, I have some questions as I am not obtaining the expected results. 

I have a TEM dataset of rat brain tissue, for which axon and myelin segmentation was performed using AxonDeepSeg. The images are in `.tif` format, of size `10.24 x 10.24 µm` and a resolution of `2048 x 2048 pixels` (i.e.  a pixel size of `0.005 x 0.005 µm`). 

Due to the characteristics of the image (e.g., contrast and definition) the segmentation outputs exhibit issues such as **holes, sharp edges, and missing axons and myelin**. To address this, we attempted to postprocess these segmentations, retaining only the blobs that appeared accurate in each mask. This approach resulted in a **sparse segmentation** dataset.

Next, we retrained the model using the sparse dataset, hoping to achieve a dense segmentation better than the one resulted from AxonDeepSeg. Unfortunately, this was not the outcome. For the retraining process, we utilized Ivadomed.

I have attached the latest configuration file I am using (having experienced with various other configurations), and I will outline some specific notes and questions that may be influencing the unsatisfactory results:  

1. By default Ivadomed uses `DiceLoss`, but after not obtaining a dense segmentation I tried to use the `GeneralizedDiceLoss` with no background. Which obtained much worse results than the base case: empty axon class for `"transfer_learning"{..., "retrain_fraction": 1.0, "reset": true}` and just small axons for `"transfer_learning"{..., "retrain_fraction": 0.3, "reset": true}`. **Is this idea right?: `no background` masks out the background from the calculation of the loss to help the model not to _penalize_ when new axons and myelin are found in the background.**

2. Has someone solved a similar task using Ivadomed and AxonDeepSeg?

Any idea or input is more than welcome :) 
Thanks in advance!

[config_microscopy_base.json](https://github.com/user-attachments/files/17328551/config_microscopy_base.json)

**Comment by mathieuboudreau** [comment]
Hi @MirenLurBarquin - I'm one of the AxonDeepSeg maintainers.

Would you be able to share a sample image and the resulting segmentation mask you get from AxonDeepSeg?

**Reply by MirenLurBarquin** [reply]
Hi @mathieuboudreau, thank you for the quick reply. 

Sure, here there is an example of the segmentation resulting from AxonDeepSeg: 

![rat01](https://github.com/user-attachments/assets/22e58c40-9f9a-4d23-859f-80d652aa3850)
![rat01_seg-axon](https://github.com/user-attachments/assets/5704efd1-5d8e-46ef-9ebd-c6a291d03454)
![rat01_seg-axonmyelin](https://github.com/user-attachments/assets/6e482f49-8c90-4560-b199-69fe275e8621)
![rat01_seg-myelin](https://github.com/user-attachments/assets/0c1b53c2-cc46-4b4f-ab19-cf55f936e8a6)

**Reply by mathieuboudreau** [reply]
Great, thanks @MirenLurBarquin !

So a few things,

* IVADOMED as a project (which was developed by the same lab as AxonDeepSeg) has been more or less discontinued, so development has progressively been winding down.
* As such, in AxonDeepSeg we've reverted back to training our own models. We recently released a major version (v5) that completely removed the IVADOMED dependency and went towards using a more generalist deep learning model, and that side of things have been mostly developped by @hermancollin 
* That being said, I just tested our most recent version with your image, and it did quite worse, see below,
![Screenshot 2024-10-10 at 11 34 43 AM](https://github.com/user-attachments/assets/621a87ed-7757-4987-810f-c16b71369f0f)

@hermancollin do you have insights on this, and potential ways to tweak the model so that it would perform closer to the IVADOMED model? Clearly the image's contrast is lower, has much more noise, and has a spatially varying luminosity.

@MirenLurBarquin - do you have some manually segmented images for this dataset?

**Reply by MirenLurBarquin** [reply]
Hi @mathieuboudreau, thank you for your help. Unluckily, we don't have any manual segmentation for this dataset.

**Reply by hermancollin** [reply]
@MirenLurBarquin would it be possible to take a look at your sparse segmentation dataset? How many images did you work with for this sparse dataset?
